### DISCUSSION TITLE
Domain Exception Error

### DISCUSSION THREAD

**Discussion by warfee** [php-aidc/label-printer#7] [Show and tell] [answered]
Hi, i try to integrate with honeywell label printer and got error
`local.ERROR:  {"exception":"[object] (DomainException(code: 0):  at /vendor/php-aidc/label-printer/src/Printer.php:37)`
I already make connection between pc and printer, pc able to reach printer when browse printer IP.
Thanks for this great package!

**Comment by jhaoda** [comment]
Please, provide code that cause this error.

**Comment by warfee** [comment]
```php
    public function test_print(){
        $printer_ip = env('PRINTER_IP');
        $label_width = env('LABEL_WIDTH');
        $label_height = env('LABEL_HEIGHT');
        $text_x_axis = env('TEXT_X_AXIS');
        $text_y_axis = env('TEXT_Y_AXIS');
        $text = env('TEXT');
        $text_font = env('FONT_TEXT');
        $text_size = env('TEXT_SIZE');

        $text_box_width = env('TEXT_BOX_WIDTH');
        $text_box_height = env('TEXT_BOX_HEIGHT');
        $text_box_border = env('TEXT_BOX_BORDER');


        $barcode_x_axis = env('BARCODE_X_AXIS');
        $barcode_y_axis = env('BARCODE_Y_AXIS');
        $barcode_data = env('BARCODE_DATA');
        $barcode_type = env('BARCODE_TYPE');
        $barcode_height = env('BARCODE_HEIGHT');
        

        $label = Label::create(Unit::MM(), $label_width, $label_height)
            ->charset(Charset::UTF8())
            ->add(Element::textBlock($text_x_axis, $text_y_axis, $text, $text_font, $text_size)->box($text_box_width, $text_box_height, $text_box_border)->anchor(Anchor::CENTER()))
            ->add(Element::barcode($barcode_x_axis, $barcode_y_axis, $barcode_data, $barcode_type)->height($barcode_height));

        (new Printer(new NetworkConnector($printer_ip)))->print($label);

        return redirect()->back()->with('execute','Test print function executed');
    }
```
this is the code, all code same as in readme but i change it to variable.

**Comment by jhaoda** [comment, marked as answer]
Use this
```php
(new Printer(new NetworkConnector($printer_ip), Compiler::create(new Fingerprint())))->print($label);
```

**Reply by warfee** [reply]
alright..i will try this one

**Comment by warfee** [comment]
There are some few error appear after use code `(new Printer(new NetworkConnector($printer_ip), Compiler::create(new Fingerprint())))->print($label);` but i try to solve the error by include library path `use PhpAidc\LabelPrinter\Compiler;
use PhpAidc\LabelPrinter\Language\Fingerprint;
use PhpAidc\LabelPrinter\Language\Fingerprint as Handlers;` and amend the code to `(new Printer(new NetworkConnector($printer_ip), Compiler::create(new Fingerprint(2))))->print($label);`. Other new error comes out `PhpAidc\LabelPrinter\Exception\CouldNotConnectToPrinter
stream_socket_client(): unable to connect to tcp://192.168.1.254:9100 (Connection timed out)`

*new Fingerprint(2) for this one i just put the value for dummy and i don't know what value use for.Please correct me if there is a mistake.

**Reply by warfee** [reply]
there are any solution that i can do for that issue?

**Reply by jhaoda** [reply]
Check your printer configuration (network settings) and so on. This is an network issue.

**Reply by warfee** [reply]
Hi, thanks alot for your response..i able to print the printer lable..project file that run locally and connect to printer..i want to ask, its that possible to command from server(if i host it to cloud) to specific printer?

**Reply by jhaoda** [reply]
You can share the printer port to the internet (very bad idea) or create VPN tunnel from cloud server to local network.

**Reply by warfee** [reply]
alright..noted! thanks alot

**Reply by warfee** [reply]
One more, got issue text or barcode not printed on the label..it that configuration label format(width n etc) or related to the computer..?
